### ISSUE TITLE
Feature request: Build for WebAssembly

### ISSUE THREAD

**Issue by Machine-Jonte** [volution/vonuvoli-scheme#105]
Hi,
Wonder if this scheme interpreter works compiling and running in native wasm (target "wasm32-unknown-unknown")?

**Comment by cipriancraciun** [comment]
I wouldn't say it's impossible to make it work, because there are a lot of "features" that gate varyous pieces of code.

However I doubt it would work without some serious effort.

(I think the main issue would be the fact that it uses the standard allocator for memory management.)

----

Could you detail perhaps your intended use-case?

**Comment by Machine-Jonte** [comment]
Yes, and thanks for the blazingly fast response! I will look into it a bit and post here if I make some progress.

**Comment by cipriancraciun** [comment]
If you want to give it a try, you can take a look at the extensive set of features, and start adding them incrementally until something breaks, or until you get what you need:

https://github.com/volution/vonuvoli-scheme/blob/b7f8d69173fe1e85cb880c6dc3e33e9e8c449748/Cargo.toml#L115-L708

For example start with the minimum of `vonuvoli_evaluator`, `vonuvoli_parser` and `vonuvoli_compiler`.  Then you can start adding `vonuvoli_values_*` and `vonuvoli_builtins_*` until it breaks.

Most likely you'll have to write a custom `main`, see the `tools_interpreter.rs` snippet below:

https://github.com/volution/vonuvoli-scheme/blob/b7f8d69173fe1e85cb880c6dc3e33e9e8c449748/sources/tools_interpreter.rs#L109-L163
